### PR TITLE
Fix server OAPIF bbox reprojection

### LINKED ISSUE #47302: Server OAPIF: bbox filter isn't working if data source CRS is not EPSG:4326/WGS84
### What is the bug or the crash?

The bounding box filter isn't working for QGIS Server OAPIF:

https://docs.qgis.org/testing/en/docs/server_manual/services/ogcapif.html#bounding-box-filter

### Steps to reproduce the issue

https://geo.kreis-viersen.de/ows/osm-daten/wfs3/collections/containerstandorte_nrw/items.json?bbox=-180,-90,180,90

or 

https://geo.kreis-viersen.de/ows/osm-daten/wfs3/collections/containerstandorte_nrw/items.json?bbox=6.365918,51.254137,6.446942,51.296776

gives no resultes but 

https://geo.kreis-viersen.de/ows/osm-daten/wfs3/collections/containerstandorte_nrw/items.json 

gives features.

### Versions

QGIS Server `master` and `3.22.3` on Ubuntu

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by elpaso**
Fix #47302

### CODE REVIEW

**Comment by m-kuhn** [comment]
I'll happily review.
Could you add a sentence to describe the reasoning behind the change?

**Comment by elpaso** [comment]
It's a bugfox

File: src/server/services/wfs3/qgswfs3handlers.cpp [hunk 1]
```diff
@@ -1231,8 +1231,7 @@ void QgsWfs3CollectionsItemsHandler::handleRequest( const QgsServerApiContext &c
 
       if ( ! filterRect.isNull() )
       {
-        const QgsCoordinateTransform ct( bboxCrs, mapLayer->crs(), context.project()->transformContext() );
-        ct.transform( filterRect );
+        const QgsCoordinateTransform ct( bboxCrs, crs, context.project()->transformContext() );
```
Comment (nyalldawson) [inline src/server/services/wfs3/qgswfs3handlers.cpp:1234]: (unrelated) Doesn't this block a try/catch QgsCsException? Or is that handled at a higher level?

Comment (elpaso) [reply src/server/services/wfs3/qgswfs3handlers.cpp:1234]: Exceptions are actually catched at a higher level but it makes sense to throw a message here, I've added it.

**Comment by elpaso** [comment]
> That's clear. I have read the title and the issue number in here. I have also clicked the link to read the issue itself. Still, a very short description giving context on what the crs change does would help me getting started with the review.

@m-kuhn this is a one-line change (actually a one-token change): previously the bbox crs which is used for filtering features was transformed to the layer's crs but since the features are transformed to the requested crs the bbox filter should have been transformed to the requested crs instead. This PR does exactly that and adds a test for that.

**Comment by elpaso** [comment]
Unrelated CI failure.

**Comment by pathmapper** [comment]
Tested locally and confirmed that this fixes the issue. Thanks @elpaso !

**Comment by m-kuhn** [comment]
> previously the bbox crs which is used for filtering features was transformed to the layer's crs but since the features are transformed to the requested crs the bbox filter should have been transformed to the requested crs instead.

Thanks 👍

**Review by m-kuhn — APPROVED** [review]

**Pull request merged**